src/services/paras/ParasService.ts

```diff
@@ -223,21 +223,47 @@ export class ParasService extends AbstractService {
 			phase: IOption<AuctionPhase>,
 			winning;
 		if (auctionInfoOpt.isSome) {
+			/**
+			 * If `AuctionInfo:::<T>:get()` is `Some`, it returns a tuple where the first item is the
+			 * lease period index that the first of the four contiguous lease periods
+			 * an auction is for. The second is the block number when the auction will
+			 * 'being to end', i.e. the first block of the Ending Period of the auction
+			 */
 			[leasePeriodIndex, beginEnd] = auctionInfoOpt.unwrap();
-			const endingOffset = this.endingOffset(blockNumber, beginEnd);
-
-			const winningOpt = endingOffset
-				? await historicApi.query.auctions.winning<Option<WinningData>>(
-						endingOffset
-				  )
-				: await historicApi.query.auctions.winning<Option<WinningData>>(
-						// when we are not in the ending phase of the auction winning bids are stored at 0
-						0
-				  );
-
-			if (winningOpt.isSome) {
+
+			/**
+			 * End of current auctions endPeriod.
+			 */
+			finishEnd = beginEnd.add(endingPeriod);
+			/**
+			 * We determine what our phase is so we can decide how to calculate our
+			 * ending offset.
+			 */
+			if (finishEnd.lte(blockNumber)) {
+				phase = 'vrfDelay';
+			} else {
+				phase = beginEnd.gt(blockNumber) ? 'startPeriod' : 'endPeriod';
```

Comment:
So for the `vrfDelay` I actually took this into account and test for it, but this is a great catch, because I actually tested off by 2 instead of 1, so I will fix that!

Reply:
fixed